src/main/java/org/javawebstack/validator/Validator.java

```diff
@@ -19,27 +20,47 @@
     private static final Map<String, Constructor<? extends ValidationRule>> validationRules = new HashMap<>();
     private static final Map<Class<?>, Validator> validators = new HashMap<>();
 
+    private static final List<Class<? extends Annotation>> RULE_ANNOTATIONS = Arrays.asList(
```

Comment:
this needs to be some kind of registry like it's done for the string based rules aswell, so you can register your own rules

Reply:
we can probably also merge the registration with string rules